gapis/perfetto/android/trace.go

```diff
@@ -240,8 +262,14 @@ func (p *Process) Capture(ctx context.Context, start task.Signal, stop task.Sign
 	// Signal that we are ready to start.
 	atomic.StoreInt64(written, 1)
 
-	if p.deferred && !start.Wait(ctx) {
-		return 0, log.Err(ctx, nil, "Cancelled")
+	if p.deferred {
```

Comment:
Could you pull this piece of logic out into a function instead of replicating the code in multiple places?